weave/scorers/scorer_types.py

```diff
@@ -79,19 +79,11 @@ class HuggingFacePipelineScorer(weave.Scorer):
         description="The device to use for the model",
         validate_default=True,
     )
+    _device: Optional["torch.device"] = PrivateAttr(default=None)
```

Comment:
This needs to be a property, otherwise it won't update properly if the user changes the device after first init